external/fv3fit/fv3fit/reservoir/readout.py

```diff
@@ -50,6 +54,59 @@ def _sort_subdirs_numerically(subdirs: Sequence[str]) -> Sequence[str]:
     return [subdir for _, subdir in sorted(zip(nums, subdirs))]
 
 
+class BatchLinearRegressor:
+    """ Solves for weights W in matrix equation AW=B
+    Where A = X_T.X + l2*I and B= X_T.y
+    """
+
+    def __init__(self, hyperparameters: BatchLinearRegressorHyperparameters):
+        self.hyperparameters = hyperparameters
+        self.A = None
+        self.B = None
+
+    def _add_bias_feature(self, X):
+        return np.concatenate([X, np.ones((X.shape[0], 1))], axis=1)
+
+    def _check_X_last_col_constant(self, X):
+        last_col = X[:, -1]
+        if len(np.unique(last_col)) > 1:
```

Comment:
It would seem a little bit more exact to check that the last column is all ones.